internal/toolexec/aspect/oncompile.go

```diff
@@ -31,9 +31,13 @@ const (
 
 var (
 	weavingSpecialCase = map[*regexp.Regexp]specialCaseBehavior{
-		regexp.MustCompile(`^github\.com/[Dd]ata[Dd]og/orchestrion(?:/.+)?$`): neverWeave,
-		regexp.MustCompile(`^gopkg\.in/DataDog/dd-trace-go.v1(?:/.+)?$`):      weaveTracerInternal,
-		regexp.MustCompile(`^github\.com/DataDog/go-tuf/client$`):             neverWeave,
+		// Carefully allow injection under the `github.com/DataDog/orchestrion/runtime` package, but
+		// nowhere else in the github.com/DataDog/orchestrion modules. The regex could be smaller if the
+		// [regexp.Regexp] supported negative lookahead... But there we are emulating it instead.
+		// See: https://regex101.com/r/kexx3q/1 (NB)
+		regexp.MustCompile(`^github\.com/[Dd]ata[Dd]og/orchestrion(?:/(?:[^r].*|r[^u].*|ru[^n].*|run[^t].*|runt[^i].*|runti[^m].*|runtim[^e].*|runtime[^/].*))?$`): neverWeave,
+		regexp.MustCompile(`^gopkg\.in/DataDog/dd-trace-go.v1(?:/.+)?$`):                                                                                           weaveTracerInternal,
+		regexp.MustCompile(`^github\.com/DataDog/go-tuf/client$`):                                                                                                  neverWeave,
```

Comment:
I believe we can simply use `strings.HasPrefix` for now right ? This regex looks like its making you sad

Reply:
I had a problem, I used regexp to try and solve the problem. I now have a `/problem/`...